src/test/scala/plugin/inline/InlineErrorCheckerTest.scala

```diff
@@ -122,11 +124,53 @@ class InlineErrorCheckerTest extends FunSuite with InlineErrorChecker with Inlin
         PredicateAccessPredicate(PredicateAccess(Seq(), calledPredId)(), FullPerm.apply()())(),
       )()
     )
+    val predicateForF = Predicate(calledPredId, Seq(), None)()
+    val program = programCopy(predicates = Seq(predicateForF))
     val pred = Predicate("loop", formalArgs = Seq(), body = maybeRecursiveBody)()
 
-    val maybeResult = nonRecursivePredsCalledBy(pred)
+    val maybeResult = nonRecursivePredsCalledBy(pred, program)
     maybeResult.fold(false) { result =>
       result.size == 1 && result(calledPredId)
     }
   }
+
+  test("predicatesCalledBy should traverse subpredicates") {
+    /*
+    predicate F(this: Ref) {
+      G(this)
+    }
+
+    predicate G(this: Ref) {
+      acc(x.f)
+    }
+
+    predicate loop(this: Ref) {
+      loop(this) && F(this)
+    }
+     */
+    val f = "F"
+    val g = "G"
+    val loop = "loop"
+    val fBody = Some(PredicateAccessPredicate(PredicateAccess(Seq(), g)(), FullPerm.apply()())())
+    val gBody = Some(FieldAccessPredicate(FieldAccess(LocalVar("x", Ref)(),Field("left", Int)(NoPosition))(), FullPerm.apply()())())
+    val loopBody = Some(
+      And(
+        PredicateAccessPredicate(PredicateAccess(Seq(), f)(), FullPerm.apply()())(),
+        PredicateAccessPredicate(PredicateAccess(Seq(), loop)(), FullPerm.apply()())(),
+      )()
+    )
+    val loopPred = Predicate("loop", formalArgs = Seq(), body = loopBody)()
+    val program = programCopy(
+      predicates = Seq(
+        loopPred,
+        Predicate(f, formalArgs = Seq(), body = fBody)(),
+        Predicate(g, formalArgs = Seq(), body = gBody)(),
+      )
+    )
+
+    val maybeResult = nonRecursivePredsCalledBy(loopPred, program)
+    maybeResult.fold(false) { result =>
+      result.size == 2 && Set(f, g).subsetOf(result)
```

Comment:
This subset check isn't necessary I can just check directly for equality. I was a fool.